### PR TITLE
did-simple: implement varint encoding

### PR DESCRIPTION
**Pull request by TheButlah**
Implements varint encoding in a const fn.

### CODE REVIEW

**Comment by TheButlah** [comment]
Marking as draft until I can write some unit tests.

File: crates/did-simple/src/varint.rs [hunk 1]
```diff
@@ -1,4 +1,82 @@
...
+pub(crate) const fn encode_varint(value: u16) -> VarintEncoding {
+	let mut out_buf = [0; 3];
+	// ilog2 can be used to get the (0-indexed from LSB position) of the MSB.
+	// We then add one to it, since we are trying to indicate length, not position.
+	let in_bit_length: u16 = if let Some(x) = value.checked_ilog2() {
+		(x + 1) as u16
+	} else {
+		return VarintEncoding {
+			buf: out_buf,
+			len: 0,
```
Comment (lyuma) [inline crates/did-simple/src/varint.rs:36]: I'm not sure if `len: 0` is a legal varint encoding of 0: you need a `0` to exist for the receiver to recognize that the buffer has ended after the first byte. An empty buffer would provide no information (unless length is encoded separately, which would defeat the whole purpose of a varint encoding in the first place)

**Review by lyuma — APPROVED** [review]
Seems ok from my limited understanding of Rust, There are three cases: 1, 2 and 3 byte length, and each of the three cases should work.

I have a comment about the 0 case, since that should fall under case 1, but it seems to return length 0 instead, and I'm not sure if that is legal within the contract of this API

File: crates/did-simple/src/varint.rs (hunk 1 above)
Comment (TheButlah) [reply crates/did-simple/src/varint.rs:36]: yeah, I think you're right

Comment (TheButlah) [reply crates/did-simple/src/varint.rs:36]: ah, we fixed this in #100 , whoops 😓

**Pull request merged**